user_modules/shared_testcases/commonStepsResumption.lua

```diff
@@ -0,0 +1,175 @@
+          type = "Choice",
+          vrCommands = {"VrChoice"}})
+  on_hmi_call:Do(function(_,data)
+    grammarIDValue = data.params.grammarID
+    Test.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS" })
+  EXPECT_NOTIFICATION("OnHashChange"):Do(function(_, data)
+    Test.currentHashID = data.payload.hashID
+  end)
+end
+
+local function CheckTimeBoundaries(time)
+  local timeout = tonumber(commonFunctions:read_parameter_from_smart_device_link_ini("ApplicationResumingTimeout"))
+  return function (exp, _)
+    if exp.occurences == 2 then
+      local time2 = timestamp()
+      local time_to_resumption = time2 - time
+      if time_to_resumption >= timeout and time_to_resumption < (timeout + 500) then
+        commonFunctions:userPrint(33, "Time to HMI level resumption is " .. tostring(time_to_resumption) ..", expected ~ " .. timeout )
+        return true
+      else
+        commonFunctions:userPrint(31, "Time to HMI level resumption is " .. tostring(time_to_resumption) ..", expected ~ " .. timeout )
+        return false
+      end
+    elseif exp.occurences == 1 then
+      return true 
+    end
+  end
+end
+
+function commonStepsResumption:RegisterApp(app, additional_expectations , resume_vr_grammars)
+  local default_additional_expectations = function (app) 
+    EXPECT_NOTIFICATION("OnHMIStatus",
+      {hmiLevel = "NONE", systemContext = "MAIN", audioStreamingState = "NOT_AUDIBLE"})
+  end
+  if (not additional_expectations) then 
+    additional_expectations = default_additional_expectations
+  end
+  local correlation_id = Test.mobileSession:SendRPC("RegisterAppInterface", app)
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+  						{ application = {appName = app.appName}, resumeVrGrammars = resume_vr_grammars}):Do(function (_, data)
+    app.hmi_app_id = data.params.application.appID
+  end)
+  Test.mobileSession:ExpectResponse(correlation_id, { success = true})  
+  local exp = additional_expectations(Test, app)
+  return exp
+end
+
+function commonStepsResumption:ExpectResumeAppFULL(app)
+  local audio_streaming_state = "AUDIBLE"
+  if(app.isMediaApplication == false) then 
+    audio_streaming_state = "NOT_AUDIBLE"
+  end
+  local time = timestamp()
+  local app_activated = EXPECT_HMICALL("BasicCommunication.ActivateApp", 
+                        {appID = Test.applications[app.appName]}):Do(function(_,data)
+    Test.hmiConnection:SendResponse(data.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
+  end)
+  EXPECT_NOTIFICATION("OnHMIStatus",
+    {hmiLevel = "NONE", systemContext = "MAIN", audioStreamingState = "NOT_AUDIBLE"},
+    {hmiLevel = "FULL", systemContext = "MAIN", audioStreamingState = "AUDIBLE"}):ValidIf(CheckTimeBoundaries(time))
+    :Do(function(_,data)
+      Test.hmiLevel = data.payload.hmiLevel
+    end):Times(2)
+  return app_activated 
+end
+
+function commonStepsResumption:ExpectResumeAppLIMITED(app)
```

Comment:
@LitvinenkoIra, LIMITED hmiLevel is applicable only for media application.app.isMediaApplication = true. You should check this directly in script for LIMITED level.